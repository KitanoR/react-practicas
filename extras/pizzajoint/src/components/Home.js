import React from 'react';
import { Link } from 'react-router-dom';
import { motion } from 'framer-motion';
import Loader from './Loader';

const buttonVariants = {
  visible: {
    x: [0, -12, 20, -20, 20, 0],
    transition: {
      delay: 2
    }
  },
  hover: {
    scale: 1.1,
    boxShadow: "0px 0px 8px rgb(255,255,255)",
    textShadow: "0px 0px 8px rgb(255, 255,255)",
    transition: {
      duration: 0.3,
      yoyo: 10
    }
  }
}

const containerVariants = {
  hidden: {
    opacity: 0
  },
  visible: {
    opacity: 1,
    transition: { delay: 1.5, duration: 1.5 }
  },
  exit: {
    x: '-100vw',
    transition: {
      ease: 'easeInOut'
    }
  }
}
const Home = () => {
  return (
    <motion.div 
      variants={containerVariants}
      initial="hidden"
      animate="visible"
      exit="exit"
      className="home container">
      <motion.h2
      >
        Welcome to Pizza Joint
      </motion.h2>
      <Link to="/base">
        <motion.button
          variants={buttonVariants}
          animate="visible"
          whileHover="hover"
        >
          Create Your Pizza
        </motion.button>
      </Link>
      <Loader />
    </motion.div>
  )
}

export default Home;

// whileHover= {{ 
//   scale: 1.1,
//   textShadow: "0px 0px 8px rgb(255, 255,255)",
//   boxShadow: "0px 0px 8px rgb(255,255,255)"
// }}