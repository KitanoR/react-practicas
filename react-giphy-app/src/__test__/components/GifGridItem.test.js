import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Pruebas <GifGridItem/>',() => {
    const titulo = 'titulo';
    const url = 'https://algo.com/algo.png'
    let wrapper = shallow(
        <GifGridItem title={titulo} url={url} />
    );

    beforeEach(() => {
        wrapper = shallow(
            <GifGridItem title={titulo} url={url} />
        )
    });
    test('Debe mostrar <GifGridItem />', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);
    });

    test('Debe de tener la imagen igual a la ur y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titulo);
    });

    test('Debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const clase = div.props().className;
        const contiene_animate = clase.includes('animate__fadeIn');
        expect(contiene_animate).toBe(true);
    });


})