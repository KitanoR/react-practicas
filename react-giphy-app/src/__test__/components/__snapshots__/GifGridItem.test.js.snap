// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem/> Debe mostrar <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://algo.com/algo.png"
  />
  <p>
     
    titulo
     
  </p>
</div>
`;
